command/connect/envoy/envoy.go

```diff
@@ -321,6 +331,27 @@ func (c *cmd) lookupProxyIDForSidecar() (string, error) {
 	return proxyCmd.LookupProxyIDForSidecar(c.client, c.sidecarFor)
 }
 
+func (c *cmd) lookupGRPCPort() (int, error) {
+	self, err := c.client.Agent().Self()
```

Comment:
This is so magical. I'd never even considered actually just asking the agent which port to use. 👍